File: 2022/julia/01.jl
# Advent of Code 2022, day 1
# (c) aichingert

module Day1

function solve(vec,to)
	return sum([vec[i] for i in 1:to])
end

inp = []

for s in split(strip(read("../input/01", String)), "\n\n")
	push!(inp, sum(map(n->parse(UInt32,n), split(s, "\n"))))
end

sort!(inp, rev=true)

println("Part 1:" * string(solve(inp,1)))
println("Part 2:" * string(solve(inp,3)))

end

File: 2022/rust/01.rs
// Advent of Code 2022, day 1
// (c) aichingert

fn solve(cal: &Vec<u32>, to: usize) -> u32 {
    (1..=to).map(|i| cal[cal.len() - i]).sum::<u32>()
}

fn main() {
    let mut inp = std::fs::read_to_string("../input/01").unwrap().trim().split("\n\n").map(|s| s.split('\n').map(|n| n.parse::<u32>().unwrap()).sum::<u32>()).collect::<Vec<u32>>();
    inp.sort();

    println!("Part 1: {}", solve(&inp,1));
    println!("Part 2: {}", solve(&inp,3));
}
